compiler/record-hessian/src/HessianObserver.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "record-hessian/HessianObserver.h"
+
+using DataType = luci_interpreter::DataType;
+
+namespace record_hessian
+{
+
+void HessianObserver::postTensorWrite(const luci::CircleNode *node,
+                                      const luci_interpreter::Tensor *tensor)
+{
+
+  auto node_outputs = loco::succs(node);
+  for (auto node_output : node_outputs)
+  {
+    auto cur_node = dynamic_cast<luci::CircleNode *>(node_output);
+    assert(cur_node != nullptr);
```

Comment:
`assert` will be nothing in release build and won't cover our SE analysis warnings.